includes/discovery/functions.inc.php

```diff
@@ -1387,6 +1387,9 @@ function find_device_id($name = '', $ip = '', $mac_address = '')
         if ($mydomain = Config::get('mydomain')) {
             $where[] = '`hostname`=?';
             $params[] = "$name.$mydomain";
+
+            $where[] = 'concat(`hostname`, \'.\', \'' . $mydomain . '\') =?';
```

Comment:
you need to use a ? for $mydomain to avoid sql injection.

Reply:
Better now?